doc/developer/filepoints.md

```diff
@@ -0,0 +1,64 @@
+# Introduction
+
+Failpoints are a way to force instrumented code to take a particular action if a given code line is reached while the failpoint is enabled. This is useful when writing tests -- for example, Materialize can be forced to panic or I/O calls can be made to return an error.
+
+Materializes uses the [fail create](https://docs.rs/crate/fail) to enable this functionality.
+
+# Intrumenting the code
+
+## Panic-only failpoints
+
+If a failpoint needs to only cause a panic, the code needs to be instrumented with the following:
```

Comment:
If you only need the failpoint to cause a panic (without an error message), use the following: